src/systems/filecoin_markets/storage_market/storage_market_actor_code.go

```diff
@@ -252,6 +245,9 @@ func (st *StorageMarketActorState_I) _rtUpdatePendingDealStatesForParty(rt Runti
 	var relevantDealIDs []deal.DealID
 	// Populate with the set of all elements in st.Deals() in which addr is one of the parties
 	// (either client or provider).
+	//
+	// Note: as an optimization, implementations may cache efficient data structures to maintain
+	// this index.
```

Comment:
No! In a real VM, that would change the operations and hence the gas cost. Define the operations here.

Reply:
(Addressed via out-of-band discussions.)